koku/koku/pg_partition.py

```diff
@@ -1066,3 +1070,281 @@ def drop_original_objects(self):
+        self.default_partition = default_partition
+        self.partition_trigger_info = partition_trigger_info or PartitionTriggerInfo(self.schema_name)
+        self.partition_name = ""
+        self.tracking_rec = None
+        self.partition_parameters = {}
+        self.tx_id = str(uuid.uuid4()).replace("-", "_")
+        self._get_default_partition_tracker_info()
+
+    def _get_default_partition_tracker_info(self):
+        sql = f"""
+SELECT *
+  FROM "{self.schema_name}"."partitioned_tables"
+ WHERE schema_name = %s
+   AND partition_of_table_name = %s
+   AND table_name = %s
+   AND partition_parameters->>'default' = 'true'
+"""
+        LOG.info(f"Getting default tracker record for {self.default_partition} of {self.partitioned_table}")
+        cur = conn_execute(sql, (self.schema_name, self.partitioned_table, self.default_partition), _conn=self.conn)
+        default_tracker = fetchone(cur)
+        self.partition_type = default_tracker["partition_type"]
+        self.partition_key = default_tracker["partition_col"]
+
+    def _update_partition_tracking_record(self, set_col):
+        LOG.debug(f"Execute SQL to update partition_parameters.{set_col}")
+        if set_col == "partition_parameters":
+            partition_parameters = self.partition_parameters.copy()
+            partition_parameters["from"] = str(partition_parameters["from"])
+            partition_parameters["to"] = str(partition_parameters["to"])
+            partition_parameters["default"] = False
+            self.tracking_rec["partition_parameters"] = json.dumps(partition_parameters)
+
+        update_sql = f"""
+UPDATE "{self.schema_name}"."partitioned_tables"
+   SET {set_col} = %s{'::boolean' if set_col == 'active' else '::jsonb'}
+ WHERE "id" = %s;
+"""
+        vals = [self.tracking_rec[set_col], self.tracking_rec["id"]]
+
+        conn_execute(update_sql, vals, _conn=self.conn)
+
+    def _create_partititon_tracking_record(self):
+        LOG.debug(f"Creating tracking record for new partition {self.partition_name}")
+        partition_parameters = self.partition_parameters.copy()
+        partition_parameters["from"] = str(partition_parameters["from"])
+        partition_parameters["to"] = str(partition_parameters["to"])
+        partition_parameters["default"] = False
+        part_track_insert_sql = f"""
+INSERT INTO "{self.schema_name}"."partitioned_tables"
+       (schema_name, table_name, partition_of_table_name, partition_type, partition_col, partition_parameters, active)
+VALUES (%s, %s, %s, %s, %s, %s::jsonb, true)
+RETURNING *;
+"""
+        vals = (
+            self.schema_name,
+            self.partition_name,
+            self.partitioned_table,
+            self.partition_type,
+            self.partition_key,
+            json.dumps(partition_parameters),
+        )
+        cur = conn_execute(part_track_insert_sql, vals, _conn=self.conn)
+        self.tracking_rec = fetchone(cur)
+
+    def _create_partition(self):
```

Comment:
`_create_partition`, `_attach_partition`, and `_detach_partition` were written to take advantage of the trigger code if the triggers are present or do it manually if not. I don't know if the triggers and trigger funcs made it into prod.